pandas/tests/frame/methods/test_replace.py

```diff
@@ -1517,18 +1517,13 @@ def test_replace_with_duplicate_columns(self, replacement):
 
         tm.assert_frame_equal(result, expected)
 
-    @pytest.mark.xfail(
-        reason="replace() changes dtype from period to object, see GH34871", strict=True
-    )
-    def test_replace_period_ignore_float(self):
-        """
-        Regression test for GH#34871: if df.replace(1.0, 0.0) is called on a df
-        with a Period column the old, faulty behavior is to raise TypeError.
-        """
-        df = DataFrame({"Per": [pd.Period("2020-01")] * 3})
+    @pytest.mark.parametrize("value", [pd.Period("2020-01"), pd.Interval(0, 5)])
+    def test_replace_ea_ignore_float(self, frame_or_series, value):
+        # GH#34871
+        df = frame_or_series([value] * 3)
         result = df.replace(1.0, 0.0)
```

Comment:
since this is now frame_or_series, can you call it `obj` instead of `df`

Reply:
Got it, done